### PR TITLE
Adds Dymos

### PR DESCRIPTION
**Pull request by sanbales**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sanbales** [comment]
Goes without saying, but I'm willing to be listed here.

File: recipes/dymos/run_test.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from os.path import join, dirname
+
+import dymos
+
+
+test_files_to_delete = [
+    # can't test these, yet, because of playwright
+    ["visualization", "linkage", "test", "test_gui.py"],
+    ["visualization", "linkage", "test", "linkage_report_ui_test.py"],
+    # unsure why this test is failing, but skipping it for now
```
Comment (JoelStansbury) [inline recipes/dymos/run_test.py:13]: Looks like it is opening a matplotlib window and blocking the test until it exits.
You can set the backend via environment variable like `set MPLBACKEND=Agg`. Maybe that would fix some things?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=682461&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=3366

Comment (JoelStansbury) [reply recipes/dymos/run_test.py:13]: doesn't explain why linux and mac are passing though

Comment (sanbales) [reply recipes/dymos/run_test.py:15]: Thanks Joel!  Testing it in [2fd1d4c](https://github.com/conda-forge/staged-recipes/pull/22430/commits/2fd1d4c09c933cbe398960188f36164946808abd)

**Comment by sanbales** [comment]
Thank you, @JoelStansbury. Got all the ✔️✔️✔️✔️✔️.

**Comment by sanbales** [comment]
@conda-forge/help-python this is ready for review

**Comment by sanbales** [comment]
forgot to add @conda-forge/staged-recipes saying this is ready for review/merge

**Comment by sanbales** [comment]
@conda-forge/staged-recipes This is ready for review.

**Comment by shb84** [comment]
Looking forward to this getting merged!

**Comment by sanbales** [comment]
@conda-forge/staged-recipes a friendly reminder this is ready for merge

File: recipes/dymos/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+source:
+  url: https://pypi.io/packages/source/d/dymos/dymos-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  entry_points:
+    - openmdao_report = dymos.linkage=dymos.visualization.linkage.report:_linkage_report_register
```
[the remark this answers is not in the document]
Comment (sanbales) [reply recipes/dymos/meta.yaml]: Thanks, Nick!  Removed this report in [9128bf8](https://github.com/conda-forge/staged-recipes/pull/22430/commits/9128bf833e2af9127cc67e0d5363f810908f3a04)

**Comment by bollwyvl** [comment]
@conda-forge/staged-recipes @conda-forge/help-python ready for review :hourglass_flowing_sand:

**Pull request merged**